app/models/reports/net_suite/donation_export.rb

```diff
@@ -3,75 +3,48 @@ module NetSuite
     class DonationExport
       include CsvExport
 
-      FIELDS = %w(tranId tranDate memo attention addressee addr1 addr2 city state zip lineId itemLine_itemRef
-                  itemLine_quantity itemLine_serialNumbers itemLine_salesPrice itemLine_amount
-                  itemLine_description).freeze
+      FIELDS = %w(donationDate donorName donorEmail addr1 addr2 city state zip
+                  donorExternalId donorExternalType memo itemName value revenueStream).freeze
 
       def each
         Donation.includes(:user, :donor, donation_details: :item).order(:id).each do |donation|
-          # Note: If the sort of the included class (DonationDetail) were done
-          # in the order() above, it would do a single query instead of 1 query
-          # for each class loaded, so use sort_by on the small set of details
-          # rather than doing a super large single query.
-          donation.donation_details.sort_by(&:id).each_with_index do |detail, i|
-            yield Row.new(donation, detail, i)
-          end
+          yield Row.new(donation)
         end
       end
 
       class Row
-        attr_reader :donation, :donation_detail, :index,
-                    :attention, :addr1, :addr2, :city, :state, :zip
+        attr_reader :donation, :donor, :addr1, :addr2, :city, :state, :zip
 
-        def initialize(donation, donation_detail, index)
+        def initialize(donation)
           @donation = donation
-          @donation_detail = donation_detail
-          @index = index
+          @donor = donation.donor
           extract_address
         end
 
-        def tran_id
-          "Donation-#{donation.id}"
-        end
+        delegate :external_id, to: :donor, prefix: true
+        delegate :external_type, to: :donor, prefix: true
+        delegate :name, to: :donor, prefix: true
+        delegate :email, to: :donor, prefix: true
```

Comment:
You can bundle these delegates together:
```
2.4.0 :001 > class A
2.4.0 :002?>   def test1; 1; end
2.4.0 :003?>   def test2; 2; end
2.4.0 :004?>   def test3; 3; end
2.4.0 :005?> end
 => :test3 
2.4.0 :006 > class B
2.4.0 :007?>   delegate :test1, :test2, :test3, to: :a, prefix: true
2.4.0 :008?>   attr_reader :a
2.4.0 :009?>   def initialize; @a = A.new; end
2.4.0 :010?> end
 => :initialize 
2.4.0 :011 > b = B.new
 => #<B:0x007faf756104a0 @a=#<A:0x007faf75610428>> 
2.4.0 :012 > b.a_test1
 => 1 
2.4.0 :013 > b.a_test2
 => 2 
2.4.0 :014 > b.a_test3
 => 3 
```

so it could be:
```
delegate :external_id, :external_type, :name, :email, to: :donor, prefix: true
```

Reply:
resolved.